sdk/typespec/typespec_client_core/src/http/request/mod.rs

```diff
@@ -291,6 +295,312 @@ impl<T> TryFrom<&'static str> for RequestContent<T> {
+            body: Bytes::from(format!(r#"{{{}}}"#, body_str)).into(),
+            phantom: PhantomData,
+        })
+    }
+}
+
+impl<T> TryFrom<HashMap<String, OffsetDateTime>> for RequestContent<T> {
+    type Error = crate::Error;
+    fn try_from(body: HashMap<String, OffsetDateTime>) -> Result<Self, Self::Error> {
+        let body_str = body
+            .into_iter()
+            .map(|(k, v)| {
+                if let Ok(formatted) = v.format(&Rfc3339) {
+                    format!(r#""{}":"{}""#, k, formatted)
+                } else {
+                    format!(r#""{}":"{}""#, k, v)
+                }
+            })
+            .collect::<Vec<_>>()
+            .join(",");
+        Ok(Self {
+            body: Bytes::from(format!(r#"{{{}}}"#, body_str)).into(),
+            phantom: PhantomData,
+        })
+    }
+}
+
+impl<T> TryFrom<HashMap<String, String>> for RequestContent<T> {
+    type Error = crate::Error;
+    fn try_from(body: HashMap<String, String>) -> Result<Self, Self::Error> {
+        let body_str = body
+            .into_iter()
+            .map(|(k, v)| format!(r#""{}":"{}""#, k, v))
+            .collect::<Vec<_>>()
+            .join(",");
+        Ok(Self {
+            body: Bytes::from(format!(r#"{{{}}}"#, body_str)).into(),
+            phantom: PhantomData,
+        })
+    }
+}
+
+impl<T> TryFrom<HashMap<String, f32>> for RequestContent<T> {
+    type Error = crate::Error;
+    fn try_from(body: HashMap<String, f32>) -> Result<Self, Self::Error> {
+        let body_str = body
+            .into_iter()
+            .map(|(k, v)| format!(r#""{}":{}"#, k, v))
+            .collect::<Vec<_>>()
+            .join(",");
+        Ok(Self {
+            body: Bytes::from(format!(r#"{{{}}}"#, body_str)).into(),
+            phantom: PhantomData,
+        })
+    }
+}
+
+impl<T> TryFrom<HashMap<String, f64>> for RequestContent<T> {
+    type Error = crate::Error;
+    fn try_from(body: HashMap<String, f64>) -> Result<Self, Self::Error> {
+        let body_str = body
+            .into_iter()
+            .map(|(k, v)| format!(r#""{}":{}"#, k, v))
+            .collect::<Vec<_>>()
+            .join(",");
+        Ok(Self {
+            body: Bytes::from(format!(r#"{{{}}}"#, body_str)).into(),
+            phantom: PhantomData,
+        })
+    }
+}
+
+impl<T> TryFrom<HashMap<String, i32>> for RequestContent<T> {
+    type Error = crate::Error;
+    fn try_from(body: HashMap<String, i32>) -> Result<Self, Self::Error> {
+        let body_str = body
+            .into_iter()
+            .map(|(k, v)| format!(r#""{}":{}"#, k, v))
+            .collect::<Vec<_>>()
+            .join(",");
+        Ok(Self {
+            body: Bytes::from(format!(r#"{{{}}}"#, body_str)).into(),
+            phantom: PhantomData,
+        })
+    }
+}
+
+impl<T> TryFrom<HashMap<String, i64>> for RequestContent<T> {
+    type Error = crate::Error;
+    fn try_from(body: HashMap<String, i64>) -> Result<Self, Self::Error> {
```

Comment:
This looks suspiciously like JSON encoding of the `HashMap`, is there possibly a more appropriate `serde_json` mechanism for encoding `HashMap`s? 

And if serde_json's serialization doesn't work, it would be nice to have a comment explaining why we couldn't use serde_json to serialize these elements.

Reply:
Thank you very much for your feedback, it's been really helpful. I addressed everything according to your suggestions - I made decimal an optional dependency and put it behind a feature, and also I am using serde_json now which has also simplified serialization code a lot. Please check out the latest iteration.